traitsui/examples/demo/Standard_Editors/VideoEditor_demo.py

```diff
@@ -0,0 +1,166 @@
+# (C) Copyright 2004-2021 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+"""
+VideoEditor
+
+Demonstrates a 'display only' video editor for Qt5+
+
+"""
+import numpy as np
+from PIL import Image
+from pyface.qt.QtGui import QImage
+from traits.api import Bool, Button, Callable, Float, HasTraits, Range, Str
+from traitsui.api import ButtonEditor, HGroup, Item, UItem, View
+from traitsui.editors.video_editor import MediaStatus, PlayerState, VideoEditor
+
+
+def QImage_from_np(image):
+    assert (np.max(image) <= 255)
+    image8 = image.astype(np.uint8, order='C', casting='unsafe')
+    height, width, colors = image8.shape
+    bytesPerLine = 4 * width
+
+    image = QImage(image8.data, width, height, bytesPerLine,
+                   QImage.Format_RGB32)
+    return image
+
+
+def np_from_QImage(qimage):
+    # Creates a numpy array from a pyqt(5) QImage object
+    width, height = qimage.width(), qimage.height()
+    channels = qimage.pixelFormat().channelCount()
+    return np.array(
+        qimage.bits().asarray(width * height * channels)
+    ).reshape(height, width, channels).astype('u1')
+
+
+def qimage_function(antialiasing=True):
+    def antialis_fun(image_fun):
+        """
+        Turns a image funciton into a QImage function,
+        bound within the viewing frames box.
+        """
+        def qimage_conv_fun(image, box_dims):
+            _np_image = image_fun(np_from_QImage(image))
+            pil_image = Image.fromarray(_np_image)
+            if antialiasing:
+                pil_image.thumbnail(box_dims, Image.ANTIALIAS)
+            else:
+                pil_image.thumbnail(box_dims)
+            _np_image = np.array(pil_image)
+            image = QImage_from_np(_np_image)
+            return image, _np_image
+        return qimage_conv_fun
+    return antialis_fun
+
+
+@qimage_function(antialiasing=False)
+def test_image_fun(image):
+    return image.transpose(1, 0, 2)
+
+
+class VideoEditorDemo(HasTraits):
+    """Defines the main VideoEditor demo class."""
+
+    #: The URL that holds the video data.
+    video_url = Str()
+
+    #: A button that
+    play_pause_button = Button()
+
+    button_label = Str('Play')
+
+    state = PlayerState
+
+    duration = Float
+
+    position = Range(low=0.0, high='duration')
+
+    error = Str
+
+    status = MediaStatus
+
+    buffer = Range(0, 100)
+
+    muted = Bool(True)
+
+    volume = Range(0.0, 100.0, value=100.0)
+
+    playback_rate = Float(1.0)
+
+    image_fun = Callable()
+
+    def _state_changed(self, new):
```

Comment:
we'd prefer not introducing new `_*_changed` methods if that's possible. can this and `_play_pause_button_changed` be rewritten with `observe`?

Reply:
Why is this?

Reply:
just because we're trying to not use `on_trait_change` and IIUC that's how the `_*_changed` methods are hooked up internally.

(Looks like `_*_changed` methods use `HasTraits._on_trait_change` which gets used by `HasTraits.on_trait_change`)